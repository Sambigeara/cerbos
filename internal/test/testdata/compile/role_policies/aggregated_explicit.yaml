# yaml-language-server: $schema=../.jsonschema/CompileTestCase.schema.json
---
mainRolePolicyFqn: "cerbos.role.admin"
