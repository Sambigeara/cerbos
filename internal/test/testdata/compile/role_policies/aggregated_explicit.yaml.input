-- derived_roles/p.yaml --
---
apiVersion: api.cerbos.dev/v1
derivedRoles:
  name: p
  definitions:
    - name: foo
      parentRoles: [user, admin]
      condition:
        match:
            expr: R.attr.foo == 42

-- derived_roles/q.yaml --
---
apiVersion: api.cerbos.dev/v1
derivedRoles:
  name: q
  definitions:
    - name: foo
      parentRoles: [user]
      condition:
        match:
            expr: R.attr.foo > 9000

-- role_policies/a.yaml --
---
apiVersion: api.cerbos.dev/v1
rolePolicy:
  derivedRole: p:foo
  rules:
    - resource: widget
      permissibleActions: [create, read]

-- role_policies/b.yaml --
---
apiVersion: api.cerbos.dev/v1
rolePolicy:
  derivedRole: q:foo
  rules:
    - resource: widget
      permissibleActions: [read, update]

-- role_policies/c.yaml --
---
apiVersion: api.cerbos.dev/v1
rolePolicy:
  role: admin
  rules:
    - resource: widget
      permissibleActions: [create, delete]
