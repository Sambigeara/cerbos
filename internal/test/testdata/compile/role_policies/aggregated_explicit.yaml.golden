{
  "fqn": "cerbos.role.admin",
  "rolePolicy": {
    "meta": {
      "fqn": "cerbos.role.admin",
      "sourceAttributes": {
        "role.admin": {
          "attributes": {
            "source": "role_policies/c.yaml"
          }
        }
      }
    },
    "role": "admin",
    "derivedRoles": {
      "p:foo": {
        "name": "foo",
        "parentRoles": {
          "admin": {},
          "user": {}
        },
        "condition": {
          "expr": {
            "original": "R.attr.foo == 42",
            "checked": {
              "referenceMap": {
                "1": {
                  "name": "R"
                },
                "4": {
                  "overloadId": [
                    "equals"
                  ]
                }
              },
              "typeMap": {
                "1": {
                  "messageType": "cerbos.engine.v1.Request.Resource"
                },
                "2": {
                  "mapType": {
                    "keyType": {
                      "primitive": "STRING"
                    },
                    "valueType": {
                      "dyn": {}
                    }
                  }
                },
                "3": {
                  "dyn": {}
                },
                "4": {
                  "primitive": "BOOL"
                },
                "5": {
                  "primitive": "INT64"
                }
              },
              "sourceInfo": {
                "location": "<input>",
                "lineOffsets": [
                  17
                ],
                "positions": {
                  "1": 0,
                  "2": 1,
                  "3": 6,
                  "4": 11,
                  "5": 14
                }
              },
              "expr": {
                "id": "4",
                "callExpr": {
                  "function": "_==_",
                  "args": [
                    {
                      "id": "3",
                      "selectExpr": {
                        "operand": {
                          "id": "2",
                          "selectExpr": {
                            "operand": {
                              "id": "1",
                              "identExpr": {
                                "name": "R"
                              }
                            },
                            "field": "attr"
                          }
                        },
                        "field": "foo"
                      }
                    },
                    {
                      "id": "5",
                      "constExpr": {
                        "int64Value": "42"
                      }
                    }
                  ]
                }
              }
            }
          }
        }
      }
    },
    "resources": {
      "widget": {
        "actions": {
          "create": {
            "permissibleForRole": {}
          },
          "delete": {
            "permissibleForRole": {}
          },
          "read": {
            "permissibleForDerivedRoles": {
              "derivedRoles": {
                "p:foo": {}
              }
            }
          }
        }
      }
    }
  },
  "compilerVersion": 1
}
